src/test/java/tech/zerofiltre/blog/domain/course/features/enrollment/EnrollIT.java

```diff
@@ -241,4 +242,55 @@ void givenCompanyUserAndCompanyCourse_enrollGetsExecutedProperly() throws Zerofi
         assertThat(enrollment.getCompanyCourseId()).isEqualTo(linkCompanyCourse.getId());
     }
 
+    @Test
+    void givenCompanyUserAndCompanyCourseWithoutCompanyId_enrollGetsExecutedProperly() throws ZerofiltreException {
```

Comment:
Pas de simulation dans un test d'intégration. Dans un test d'intégration, les seuls cas où l'on utilise des simulacres sont les appels à des services externes tel que API Rest externes, Bases de données etc. Et bien des fois, l'on utilise des simulacres qui se rapprochent assez bien du composant réel. Exemple Base h2 au lieu de base Mysql.

Résumé: pas de Mock sur nos composants JAVA dans un test d'intégration, sauf si ces composants font appel à une API Rest. Sachant que dans le cas du composant externe mysql, nous avons une base in memory H2.